assets/js/modules/analytics-4/components/settings/AdsConversionIDSettingsNotice.js

```diff
@@ -45,6 +47,24 @@ export default function AdsConversionIDSettingsNotice() {
 	const adsConversionIDMigratedAtMs = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS_4 ).getAdsConversionIDMigratedAtMs()
 	);
+	const viewContext = useViewContext();
+	const trackDismissNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'dismiss_notification'
+		);
+	};
+	const trackConfirmNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'confirm_notification'
+		);
+	};
+
+	// Track a view_notification event.
+	useMount( () => {
+		trackEvent( `${ viewContext }_GA_Ads_redirect`, 'view_notification' );
+	} );
```

Comment:
While this looks alright, however, as the notice appears conditionally, i.e. only if it hasn't been over 28 days after the migration, this `trackEvent` call isn't taking this condition into consideration.

In a scenario where the migration happened over 28 days ago and the notice has stopped appearing, this `trackEvent` call still takes place, which is something I think we don't want.

We either need to check for visibility of the notice in the viewport, or take the conditions into consideration before we call this.

Let me know what you think, thanks!

Reply:
Thank you @nfmohit I have updated this and included a conditional wrapper which is the opposite of the return null conditions we have prior to component render, so the tracking event will only occur if the migration has happened and if it was within the 28 day window.